### ISSUE TITLE
Make app more accurately portray the color of the light

### ISSUE THREAD

**Issue by dudeofawesome** [eliotstocker/Light-Controller#8]
To do this, we would store the color we last set the light to, assume it hasn't changed since then, and set the UI to match that.  This should include setting the location of the knob on the color ring.

**Comment by mrwhale** [comment]
@applamp_api or @limitlessLED they should have a grasp on the firmware. Would be awesome if we could get a hold of the source for this (hopefully its running a small linux/unix distro, would make it nice and customisable to include a database to hold info).  If anyone is keen to do it themselves, https://github.com/riptidewave93/RFLED-Server

Modify and RPi to emulate the wifibridge, and all you'd then need to do in install a database to keep light state/color etc, and make that info available via http etc etc

**Comment by nagelm** [comment]
When I was trawling the web looking for info on this hardware about 8 months ago I came across  [Futlight](http://www.futlight.com/) who as best I can tell are the manufacturers for all of these wifi lights and the makers of the MiLight software. I have seen comments on other forums where Stijn & Hamish post [Geekzone.co.nz; Ninjablocks forums] but noone has ever confirmed my suspicions (and fair enough too if they are trying to protect their source). 

I gave up looking for info on the transceivers which the bridge/bulbs use. But your idea is much simpler (and more likely to work), I like it. Would be very interested to hear what Stijn & Hamish have to add otherwise I'll see what I can find out.

**Comment by mrwhale** [comment]
For those interested, this is all I could find on the matter
Someone modded their pi to work as the bridge, so sortof finds out what hardware is used:
http://servernetworktech.com/2014/09/limitlessled-wifi-bridge-4-0-conversion-raspberry-pi/

This guy makes a youtube vid for the above blog:
https://www.youtube.com/watch?v=M_swGkuTqw4

Some guy in the comments of above finds out hardware deets of wifi bridge:
http://www.szkexinda.com/Upload/PicFiles/2014.2.8_9.21.48_1259.pdf

**Comment by AppLamp-API** [comment]
Yeah FUTLIGHT is the manufacturer, but they also use other chinese 
companies for the tech... so they outsource all engineering... that 
really sucks because we have a lot of improvement ideas, and we never 
get there.

**Comment by eliotstocker** [comment]
@AppLamp-API you dont have a wifi bridge without it clothes on you could send us a picture of do you?
if so we may see some JTAG connectors with which we might get into the Software

**Comment by AppLamp-API** [comment]
Vriendelijke groet,
Stijn De Ryck
www.applamp.nl <http://www.applamp.nl>
T: +31(0)307370522

[logo] <http://www.applamp.nl>

The WiFi board used is HF-LPT100
http://www.hi-flying.com/products_detail/&productId=d7fa2397-c1ee-4e4f-89c3-600a864861de&comp_stats=comp-FrontProducts_list01-111.html
See the documents section on that page:
Data Sheet

　Title 	*　Description* 	
Version
	
Time
　HF-LPT100 User Manual 
<http://www.hi-flying.com/downloadsfront.do?method=picker&flag=all&id=253ccee8-9a9e-45dd-933d-19e5dd607c1d&fileId=9dae0180-3725-47ba-85c5-a91d7131eef5>
	　HF-LPT100 User Manual 	
1.7
	
2014.10.14
　HF-LPT100_EVK_Design_Package 
<http://www.hi-flying.com/downloadsfront.do?method=picker&flag=all&id=b31cef59-2088-4284-bf29-4afe5212bcb1&fileId=d3fbfd1e-00c9-4003-8b86-0701718182df>
	　HF-LPT100_EVK_Design_Package 	
1.2
	
  2014.04.16
　EVK_Quick_Start_Guide 
<http://www.hi-flying.com/downloadsfront.do?method=picker&flag=all&id=a974580c-9a9b-4329-a554-6bd54aa8500d&fileId=b545c662-4ec7-49a4-aea4-e0997f062a62>
	　EVK_Quick_Start_Guide 	1.0 	2013.12.23
　Mass Production Upgrade Tool 
<http://www.hi-flying.com/downloadsfront.do?method=picker&flag=all&id=6c5e27a4-2199-4314-8b52-131d05013b50&fileId=07bc0a59-0a0d-4fb4-a5e5-c3403f09ab08>
	　Mass Production Upgrade Tool 	1.0 	2014.03.06
　Customized webpage manual 
<http://www.hi-flying.com/downloadsfront.do?method=picker&flag=all&id=6c5e27a4-2199-4314-8b52-131d05013b50&fileId=07bc0a59-0a0d-4fb4-a5e5-c3403f09ab08>
	　Customized webpage manual 	1.0.1 	2014.10.22
　WiFi connection guideline 
<http://www.hi-flying.com/downloadsfront.do?method=picker&flag=all&id=1e300973-a434-4e26-8151-df498e3485d8&fileId=2bfa924c-9805-445f-8bbb-e7aa846a362c>
	　WiFi connection guideline 	1.0 	2014.12.09
　HF LP Series pins mapping table 
<http://www.hi-flying.com/downloadsfront.do?method=picker&flag=all&id=94788c61-8a61-443e-b2db-7d0252f79bab&fileId=fba58796-9918-4c82-9ae5-16221b9f4842>
	    HF LP Series pins mapping table 	1.0 	2014.12.09



For RF also the PL1167 is used
I don't see dedicated JTAG pins but see for yourself,  it's easy to open 
on the bottom with a fine screwdriver inbetween the plastic!

Cheers,
Stijn
